### PR TITLE
Add classroom timepicker

### PR DESCRIPTION
**Pull request by jfdoming**
## Notion ticket link

[Add start / end time in classrooms](https://www.notion.so/uwblueprintexecs/Add-start-end-time-in-classrooms-dfa987cf06384dcd8cdc28333270bd36)

## Implementation description
Update the date picker in the classroom modal to be a DateTimePicker. This PR also includes a bunch of minor UI fixes related to this change; have a look at the commits for a full list.

image
image

## Steps to test
1. Open the add classroom modal.
2. Validate the time looks good and validation still occurs.
3. Save a classroom and verify editing still works.

## What should reviewers focus on?
* 

## Checklist
- [x] My PR name is descriptive and in imperative tense
- [x] My commit messages are descriptive and in imperative tense. My commits are atomic and trivial commits are squashed or fixup'd into non-trivial commits
- [x] I have run the appropriate linter(s)
- [x] I have requested a review from the PL, as well as other devs who have background knowledge on this PR or who will be building on top of this PR

### CODE REVIEW

File: frontend/src/components/common/form/TimePicker.tsx [hunk 1]
```diff
@@ -34,6 +34,9 @@ const chakraStyles = (
       outline: "none",
     },
     border: "none",
+    _disabled: {
+      opacity: 1,
```
Comment (carissa-tang) [inline frontend/src/components/common/form/TimePicker.tsx:38]: what's this for?

**Review by carissa-tang — APPROVED** [review]
looks good (i changed base branch for easier review - you can revert)

File: frontend/src/components/common/form/TimePicker.tsx (hunk 1 above)
Comment (jfdoming) [reply frontend/src/components/common/form/TimePicker.tsx:38]: Due to an implementation detail the `opacity`-based styling for disabled elements was getting applied twice, meaning that this component looked different than other disabled components. Technically this fix belongs in the previous pr but that's already merged so I thought this was fine.

**Comment by jfdoming** [comment]
(force push just to rebase on previous PR)

**Pull request merged**